simulator/func_sim/alu.h

```diff
@@ -158,27 +158,47 @@ struct ALU
     template<typename I, typename T> static void riscv_subtraction( I* instr)  { instr->v_dst = sign_extension<bitwidth<T>, typename I::RegisterUInt>(narrow_cast<T>( instr->v_src1) - narrow_cast<T>( instr->v_src2)); }
     template<typename I, typename T> static void addition_imm( I* instr) { instr->v_dst = narrow_cast<T>( instr->v_src1) + narrow_cast<T>( instr->v_imm); }
 
+    template<typename T, typename T_src1, typename T_src2> static
+    auto test_addition_overflow( T_src1 src1, T_src2 src2)
+    {
+        typedef typename std::make_signed<T_src1>::type T_src1_signed;
+        typedef typename std::make_signed<T_src2>::type T_src2_signed;
+        typedef typename std::make_signed<T>     ::type T_signed;
```

Comment:
We have our own macro of `sign_t` for that. It handles 128 bit variables as well, please use it.
Also, please use modern C++ type definition which is more clear:
```c++
using T_signed = sign_t<T>;
```